### PR TITLE
[TF-928, TF-1113]: Visuals Screen to customise Overlay cursor

### PR DESCRIPTION
**Pull request by JordanBerni-UL**
## Summary

Add the Visuals screen to customise the Overlay app cursor. This is one of the last features needed to bring Web Settings up to parity with Unity.
NOTE: This screen (and tab) only appears when inside a Tauri environment as it relies on access to the file system.
Also includes a change to Overlay App to recognise 2 new pre-sets (Outline Light and Outline Dark)

https://ultrahaptics.atlassian.net/browse/TF-928
https://ultrahaptics.atlassian.net/browse/TF-1113

### Tests Added

Added https://ultrahaptics.atlassian.net/projects/TF?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/TF-T441
and
https://ultrahaptics.atlassian.net/projects/TF?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/project/15206/testCase?folderId=7310977

## Contributor Tasks

_These tasks are for the pull request creator to tick off._

- [x] PO review (optional depending on work type)
- [ ] XDR review (optional depending on work type)
- [x] QA review (or another developer if no QA is available)
- [ ] ~~Ensure documentation requirements are met e.g., public API is commented~~
- [ ] ~~Relevant changelogs have been updated with user-visible changes~~
    - [ ] ~~[TouchFree Windows](/ultraleap/touchfree/blob/-/CHANGELOG-windows.md)~~
    - [ ] ~~[TouchFree BrightSign](/ultraleap/touchfree/blob/-/CHANGELOG-brightsign.md)~~
    - [ ] ~~[TouchFree Web Tooling](/ultraleap/touchfree/blob/-/TF_Tooling_Web/CHANGELOG.md)~~
- [x] Consider any licensing/other legal implications e.g., notices required

If there is an associated JIRA issue:
- [x] Include a link to the JIRA issue in the summary above
- [x] Make sure the fix version on the issue is set correctly

## Reviewer Tasks

_Add any instructions or tasks for the reviewer such as specific test considerations before this can be merged._

- [x] Developer testing
- [x] Code reviewed
- [ ] Non-code assets reviewed
- [ ] Documentation reviewed - includes checking documentation requirements are met and not missing e.g., public API is commented

### CODE REVIEW

File: TF_Settings_Web/src/Components/Header/TabSelector/TabSelector.module.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
+@use '@/variables.module.scss' as v;
+
+.tab {
+    --tab-bg: transparent;
+    --tab-bg-active: #{v.$app-background-color};
+
+    &--light {
+        --tab-bg: #{v.$app-background-color};
+        --tab-bg-active: #{v.$dark-grey-gradient};
+    }
```
Comment (cartmouse) [inline TF_Settings_Web/src/Components/Header/TabSelector/TabSelector.module.scss:10]: :+1: This is a really nice pattern!

File: TF_Settings_Web/src/Pages/Visuals/ColorPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                {cursorSections.map((name, index) => (
+                    <TabSelector
+                        name={name}
+                        key={index}
+                        tabIndex={index}
+                        scheme={'light'}
+                        forceHideDivider={index === cursorSections.length - 1}
+                        activeTabIndex={activeTabIndex}
+                        setAsActiveTab={setActiveTabIndex}
+                    />
```
Comment (cartmouse) [inline TF_Settings_Web/src/Pages/Visuals/ColorPicker.tsx:33]: :+1: Love that this is now reusable

File: TF_Settings_Web/package.json [hunk 3]
```diff
@@ -10,8 +10,11 @@
         "react": "^17.0.2",
         "react-canvas-confetti": "^1.3.0",
         "react-circular-progressbar": "^2.1.0",
+        "react-colorful": "^5.6.1",
         "react-dom": "^17.0.2",
+        "rgb-hex": "^4.0.0",
```
Comment (cartmouse) [inline TF_Settings_Web/package.json]: 🙈 This doesn't seem to be used anywhere, can we remove?

File: TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        if (!section) return;
+
+        section.setProperty('--center-fill', cursorStyle[0]);
+        section.setProperty('--outer-fill', cursorStyle[1]);
+        section.setProperty('--center-border', cursorStyle[2]);
+    };
+
+    const currentPreset = useMemo((): CursorPreset => {
+        return presets[state.activeCursorPreset];
+    }, [state.activeCursorPreset]);
```
Comment (cartmouse) [inline TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx:79]: :shrug: Could be one line without the `return`

File: TF_Settings_Web/package.json (hunk 3 above)
Comment (JordanBerni-UL) [reply TF_Settings_Web/package.json]: Good catch - was let in from trying out a different solution

File: TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx (hunk 4 above)
Comment (JordanBerni-UL) [reply TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx:79]: Puts it all on one line which feels less readable

**Review by cartmouse — APPROVED** [review]

**Pull request merged**